### PR TITLE
Simplified event detection documentation

### PR DESCRIPTION
**Pull request by JoseGuzman**
A suggestion to simplify the event detection procedure

### CODE REVIEW

File: src/stimfit/py/minidemo.py [hunk 1]
```diff
@@ -1,43 +1,68 @@
-"""Performs fits as decribed in the manual to create 
-preliminary and final templates from minis.dat.
-last revision: May 09, 2008
-C. Schmidt-Hieber
 """
+This script set base, peak and fit cursors to
```
Comment (neurodroid) [inline src/stimfit/py/minidemo.py]: -> This script **sets** base, ...

File: src/stimfit/py/minidemo.py [hunk 2]
```diff
@@ -1,43 +1,68 @@
-"""Performs fits as decribed in the manual to create 
-preliminary and final templates from minis.dat.
-last revision: May 09, 2008
-C. Schmidt-Hieber
 """
+This script set base, peak and fit cursors to 
+perform events detection as decribed in the Stimfit manual [1]
```
Comment (neurodroid) [inline src/stimfit/py/minidemo.py]: -> event detection

File: src/stimfit/py/minidemo.py [hunk 3]
```diff
@@ -1,43 +1,68 @@
-"""Performs fits as decribed in the manual to create 
-preliminary and final templates from minis.dat.
-last revision: May 09, 2008
-C. Schmidt-Hieber
 """
+This script set base, peak and fit cursors to 
+perform events detection as decribed in the Stimfit manual [1] 
+It ctimeates a preliminary and final templates from a file 'minis.dat.
```
Comment (neurodroid) [inline src/stimfit/py/minidemo.py]: -> It creates
-> Close quotation mark after 'minis.dat'

File: src/stimfit/py/minidemo.py [hunk 4]
```diff
@@ -1,43 +1,68 @@
...
+
+[1] https://neurodroid.github.io/stimfit/manual/event_extraction.html
 
+"""
 import stf
+
+from wx import MessageBox
+
+if stf.get_filename()[-9:] != 'minis.dat':
+    MessageBox('minis.dat is not current file!', 'Warning')
```
Comment (neurodroid) [inline src/stimfit/py/minidemo.py]: -> Use minis.dat for this demo

File: src/stimfit/py/minidemo.py [hunk 5]
```diff
@@ -1,43 +1,68 @@
...
     return stf.leastsq(5)
 
 def final():
-    """Creates a final template"""
-    stf.set_peak_start(100)
-    stf.set_peak_end(599)
-    stf.set_fit_start(100)
-    stf.set_fit_end(599)
+    """
+    Set peak, base and fit cursors around a synaptic event
```
Comment (neurodroid) [inline src/stimfit/py/minidemo.py]: -> Sets

File: src/stimfit/py/minidemo.py [hunk 6]
```diff
@@ -1,43 +1,68 @@
...
+    and performs a biexponetial fit to create the final template
```
Comment (neurodroid) [inline src/stimfit/py/minidemo.py]: -> biexponential

File: doc/sphinx/manual/event_extraction.rst [hunk 7]
```diff
@@ -7,19 +7,19 @@ Event extraction by template matching
...
+`Stimfit <http://www.stimfit.org>`_ can analize spontaneous events such as EPSCs or EPSPs using a template matching algorithm (described by Jonas et al. (1993) [#Jonas1993]_and with some implementations from Clemens and Bekkers (1997) [#ClemensBekkers1997]_). A template is a waveform :math:`p(t)` of length :math:`n` that represents the time course of a typical or exemplary event. The template is slid over the recording trace :math:`r(t)`, and at each sampling point (:math:`s`) is multiplied by a scaling factor :math:`m` and an offset :math:`c` is added or subtracted so that the sum of squared errors :math:`\chi^2(t_s)` between the trace and the template is minimized:
```
Comment (neurodroid) [inline doc/sphinx/manual/event_extraction.rst]: analize -> analyze

Comment (neurodroid) [inline doc/sphinx/manual/event_extraction.rst]: Missing space after the Jonas 1993 citation

Comment (neurodroid) [inline doc/sphinx/manual/event_extraction.rst]: There are some more grammar issues (e.g. missing subject etc.) in this sentence. Are there any mistakes in the original version that would require this revision?

File: doc/sphinx/manual/event_extraction.rst [hunk 8]
```diff
@@ -33,17 +33,17 @@ In practice, the following steps need to be performed to extract events with `St
...
 
-Create a preliminary template
------------------------------
+1. Create a preliminary template
+--------------------------------
 
-In general, the template waveform :math:`p(t)` can be of arbitrary shape. A typical way of creating such a template is to fit a function with a time course matching the event kinetics to some exemplary events. For example, EPSCs can typically be modeled with the sum or the product of two exponential functions [#f1]_. In practice, a robust estimate for a template can be obtained using an iterative approach, which will be illustrated here using a recording of miniature EPSCs that you can download `here <http://stimfit.org/tutorial/minis.dat>`_.
+A typical way of creating template is to fit a exemplary or typical event to a model with similar kinetics. For example, EPSCs can typically be modeled with the sum or the product of two exponential functions [#f1]_. In practice, a robust estimate for a template can be obtained fiting an event to a biexponential function, selecting the detected events and repeating this operation iteratively. You can try this iterative approach with a file that you can download `here <http://stimfit.org/tutorial/minis.dat>`_.
```
Comment (neurodroid) [inline doc/sphinx/manual/event_extraction.rst]: Missing article in front of "template"

Comment (neurodroid) [inline doc/sphinx/manual/event_extraction.rst]: "fiting" -> "by fitting"

File: doc/sphinx/manual/event_extraction.rst [hunk 9]
```diff
@@ -64,15 +64,15 @@ We now use the bait example to fish some more large and isolate events. Choose "
...
 
 
     .. figure:: images/events.png
         :align: center
 
         **Fig. 23:**  Detected events.
 
-To view the isolated events in a new window, you have to switch to the event editing mode, either by pressing **E** or by activating the corresponding button in the toolbar (Fig. 24). When you now click on the trace with the right mouse button, a menu will show up. Select "Extract selected events" from this menu. this will put the exemplary EPSCs into a new window.
+To get the detected events in a new window, switch to the event editing modeby pressing **E** or by activating the corresponding button in the toolbar (Fig. 24). Clicking on the trace with the right mouse button will allow a menu to appear. Select "Extract selected events" from this menu to put the exemplary events into a new window.
```
Comment (neurodroid) [inline doc/sphinx/manual/event_extraction.rst]: Missing space between "mode" and "by". "get events in a new window" is unclear.

File: doc/sphinx/manual/event_extraction.rst [hunk 10]
```diff
@@ -82,38 +82,38 @@ To view the isolated events in a new window, you have to switch to the event edi
...
-We now create the average of all extracted events, as explained in chapter 1. Then, we fit a biexponential function to the average, as explained above for the single EPSC. Remember to set the baseline, peak and fit window cursors appropriately before performing the fit, and to update all calculations. Again, you can make use of a function from the ``minidemo`` module to set the cursors and perform the fit:
+To create the final template, we need to get the average of all detected events and fit it to a biexponential function. Remember to set the baseline, peak and fit window cursors appropriately before performing the fit, and to update all calculations. Again, you can make use of a function from the ``minidemo`` module to set the cursors and perform the fit:
 
 ::
 
     >>> import minidemo # if you have not imported it already
     >>> minidemo.final()
 
-The final template should look similar as shown in Fig. 25
+The final template should look similar as shown in Fig. 25 and give timecontants of 0.13 and 15.36 ms respectively.
```
Comment (neurodroid) [inline doc/sphinx/manual/event_extraction.rst]: "timecontants" -> "time constants"

File: doc/sphinx/manual/event_extraction.rst [hunk 11]
```diff
@@ -82,38 +82,38 @@ To view the isolated events in a new window, you have to switch to the event edi
...
 
         **Fig. 25:** Creating a final template.
 
-Extract all events
-------------------
+4. Extract all events
+---------------------
 
-Go back to the original file (minis.dat). Extracting all events with the final template is done in nearly the same way as described above for the preliminary template. However, you have to choose the correct template in the event dialog: The final template in this case is the second on the list (Fig. 26). For this final run, we will lower the detection threshold to a value of 3, as suggested by Clements and Bekkers (1997).
+In the original file (minis.dat) we will extrac all events with the final template. Similarly as before, select in Analysis->Event detection -> Template matching, but now the final template is the second on the menu list (Fig. 26). For this final run, we will lower the detection threshold to a value of 3, as suggested by Clements and Bekkers (1997).
```
Comment (neurodroid) [inline doc/sphinx/manual/event_extraction.rst]: "extrac" -> "extract"

File: doc/sphinx/manual/event_extraction.rst [hunk 12]
```diff
@@ -82,38 +82,39 @@ To view the isolated events in a new window, you have to switch to the event edi
...
+5. Edit detected events
+-----------------------
+
+The detected events have to be visually inspected to remove false-positives and add false-negatives. Remove false-positives with the unselected the checkbox next to the arrow indicating an event (Fig. 23). To add false-negatives, switch to the event-editing mode (Fig. 24) and right-click on the trace at the position where the event starts (Fig. 27). To screen the whole trace, it is convenient to use **Shift**  and left arrow at the same time. Once you are done with editing, choose "Extract selected events" from the context menu.
```
Comment (neurodroid) [inline doc/sphinx/manual/event_extraction.rst]: "Remove false-positives with the unselected the checkbox next to the arrow indicating an event", please correct

File: src/stimfit/gui/dlgs/eventdlg.cpp [hunk 13]
```diff
@@ -103,18 +127,29 @@ wxDialog( parent, id, title, pos, size, style ), m_threshold(4.0), m_mode(stf::c
...
 
 void wxStfEventDlg::EndModal(int retCode) {
+    wxCommandEvent unusedEvent;
     // similar to overriding OnOK in MFC (I hope...)
-    if (retCode==wxID_OK) {
+    switch( retCode) {
+
+    case wxID_OK:
         if (!OnOK()) {
+            wxLogMessage(wxT("Please, select a detection method"));
```
Comment (neurodroid) [inline src/stimfit/gui/dlgs/eventdlg.cpp]: Delete "Please, "

File: doc/sphinx/manual/event_extraction.rst [hunk 14]
```diff
@@ -7,19 +7,19 @@ Event extraction by template matching
...
+`Stimfit <http://www.stimfit.org>`_ can analyze spontaneous events such as EPSCs or EPSPs using a template matching algorithm (described by Jonas et al. (1993) [#Jonas1993]_ and with some implementations from Clemens and Bekkers (1997) [#ClemensBekkers1997]_). A template is a waveform :math:`p(t)` of length :math:`n` that represents the time course of a typical or exemplary event. The template is slid over the recording trace :math:`r(t)`, and at each sampling point (:math:`s`) is multiplied by a scaling factor :math:`m`, and an offset :math:`c` is added or subtracted so that the sum of squared errors :math:`\chi^2(t_s)` between the trace and the template is minimized:
```
Comment (neurodroid) [inline doc/sphinx/manual/event_extraction.rst]: " and with some implementations from Clemens and Bekkers (1997) " does not make sense. There's only one implementation of the code.

Comment (neurodroid) [inline doc/sphinx/manual/event_extraction.rst]: Some words are missing in the revision. See original text.

**Comment by JoseGuzman** [comment]
reverted to initial event extraction page and remove Please from wxLogMessage

**Pull request merged**